src/SSW.Rewards.Application/Quizzes/Commands/SubmitUserQuiz/SubmitUserQuizCommand.cs

```diff
@@ -72,10 +73,11 @@ private void AddQuizAchievement(int achievementId, int userId)
                 AwardedAt     = DateTime.UtcNow,
                 AchievementId = achievementId
             };
+            
             _context.UserAchievements.Add(quizCompletedAchievement);
         }
 
-        private async Task AddCompletedQuiz(int quizId, int userId, List<QuizAnswerDto> answers, bool passed)
+        private void AddCompletedQuiz(int quizId, int userId, List<QuizAnswerDto> submittedAnswers, IEnumerable<QuizAnswer> dbAnswers, bool passed)
```

Comment:
why no `async` anymore?

Reply:
In the previous version of this method, the IDs of the answers the user had submitted were used to fetch the corresponding question from the database. In this version, the questions are passed in as they are already loaded in the original query.